Multiple definition linker error when compiling with GCC 10 and Clang 11

### Greenplum version or build
master (7.0 alpha as of commit 36d0da913481b), but this is very likely applicable to back branches as well

### autoconf options used ( config.status --config )
`./configure CC=gcc-10`
Similarly `./configure CC=clang-11`

Update: this can also happen if you turn on `-fno-common`:
`./configure CFLAGS=-fno-common`

### Expected behavior
The postgres executable compiles and links

### Actual behavior

```
pivotal@b39f85f4d1b1:~/gpdb.build.gcc10$ make -s
/usr/bin/ld.gold: error: cdb/cdbvars.o: multiple definition of 'Debug_resource_group'
/usr/bin/ld.gold: utils/misc/guc_gp.o: previous definition here
/usr/bin/ld.gold: error: cdb/cdbvars.o: multiple definition of 'gp_encoding_check_locale_compatibility'
/usr/bin/ld.gold: utils/misc/guc_gp.o: previous definition here
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:72: postgres] Error 1
make[1]: *** [Makefile:38: all-backend-recurse] Error 2
make: *** [GNUmakefile:11: all-src-recurse] Error 2

```

You'll get almost exactly the same error with Clang 11

### Step to reproduce the behavior
`./configure CC=gcc-10 && make -s`

--- a/src/backend/cdb/cdbvars.c
+++ b/src/backend/cdb/cdbvars.c
@@ -73,7 +73,7 @@ bool		Debug_print_prelim_plan;	/* Shall we log plan before adding
 
 bool		Debug_print_slice_table;	/* Shall we log the slice table? */
 
-bool		Debug_resource_group;	/* Shall we log the resource group? */
+bool		Debug_resource_group = false;	/* Shall we log the resource group? */
 
 bool		Debug_burn_xids;
 

--- a/src/backend/utils/misc/guc_gp.c
+++ b/src/backend/utils/misc/guc_gp.c
@@ -129,7 +129,6 @@ bool        test_AppendOnlyHash_eviction_vs_just_marking_not_inuse = false;
 bool		Debug_appendonly_print_datumstream = false;
 bool		Debug_appendonly_print_visimap = false;
 bool		Debug_appendonly_print_compaction = false;
-bool		Debug_resource_group = false;
 bool		Debug_bitmap_print_insert = false;
 bool		Test_print_direct_dispatch_info = false;
 bool        Test_print_prefetch_joinqual = false;
@@ -194,9 +193,6 @@ int			Debug_dtm_action_protocol = DEBUG_DTM_ACTION_PROTOCOL_DEFAULT;
 
 int			Debug_dtm_action_segment = DEBUG_DTM_ACTION_SEGMENT_DEFAULT;
 int			Debug_dtm_action_nestinglevel = DEBUG_DTM_ACTION_NESTINGLEVEL_DEFAULT;
-
-/* Enable check for compatibility of encoding and locale in createdb */
-bool		gp_encoding_check_locale_compatibility;
 
 int			gp_connection_send_timeout;
 

--- a/src/include/cdb/cdbvars.h
+++ b/src/include/cdb/cdbvars.h
@@ -675,6 +675,9 @@ extern bool gp_cte_sharing;
 /* Enable RECURSIVE clauses in common table expressions */
 extern bool gp_recursive_cte;
 
+/* Enable check for compatibility of encoding and locale in createdb */
+extern bool gp_encoding_check_locale_compatibility;
+
 /* Priority for the segworkers relative to the postmaster's priority */
 extern int gp_segworker_relative_priority;
 

